internal/api/utils.go

```diff
@@ -265,56 +260,109 @@ func TransformComponentUnits(jsonData datatypes.JSON) map[string]interface{} {
 			if math.Abs(cpuInCores) < 1 {
 				cpuInMillicores := cpuInCores * 1000
 				cpu["amount"] = math.Round(cpuInMillicores) // millicore values are rounded & don't require decimal precision
-				cpu["format"] = "millicores"
+				cpu["format"] = "m"
 			} else {
 				cpu["amount"] = truncateToThreeDecimalPlaces(cpuInCores)
-				cpu["format"] = "cores"
+				cpu["format"] = nil
 			}
 		}
 		return nil
 	}
 
+	// Current
+	current_config, ok := data["current"].(map[string]interface{})
+	if !ok {
+		log.Error("current not found in JSON")
+	}
+
+	for _, section := range []string{"limits", "requests"} {
+
+		sectionObject, ok := current_config[section].(map[string]interface{})
+		if ok {
+			memory, ok := sectionObject["memory"].(map[string]interface{})
+			if ok {
+				err := convertMemory(memory)
+				if err != nil {
+					fmt.Printf("error converting memory in %s: %v\n", sectionObject, err)
+					continue
+				}
+			}
+			cpu, ok := sectionObject["cpu"].(map[string]interface{})
+			if ok {
+				err := convertCPU(cpu)
+				if err != nil {
+					fmt.Printf("error converting cpu in %s: %v\n", sectionObject, err)
+					continue
+				}
+			}
+		}
+	}
+
 	/*
 		Recommendation data is available for three periods
+		under cost and performance keys
 		For each of these actual values will be present in
 		below mentioned dataBlocks > request and limits
 	*/
 
-	for _, period := range []string{"long_term", "medium_term", "short_term"} {
-		intervalData, ok := durationBased[period].(map[string]interface{})
+	// Recommendations
+	recommendation_terms, ok := data["recommendation_terms"].(map[string]interface{})
+	if !ok {
```

Comment:
Does this mean we don't have any recommendations? so in that case should we still go ahead and execute next for loop on "short_term", "medium_term", "long_term"?

Reply:
Am I wrong or not? Request to check?

Reply:
This observation is correct, further processing is unnecessary at this point

Reply:
not sure how I missed the return statement, poor me sometimes :\

Reply:
It wasn't there, I added it :smile: